outlines/packages.md

```diff
@@ -0,0 +1,32 @@
+# Building a package
+
+1. Why you might want to build a package
+    1. You have a package-based app
+    2. You want to publish some code for people to use in the community
+    3. You want to share code within your organization between different apps
+2. Creating a package using the command line tool
+3. Parts of a Meteor package
+    1. What different parts of package.js mean, point to the docs
+    2. Depending on other packages and build plugins
+         1. You can't depend on a specific version of Meteor; that's not what versionsFrom does
+    3. Adding files and assets
+    4. About architectures
+    5. Semver and the constraint solver
+    6. Cordova plugins
+    7. NPM packages
```

Comment:
I agree it should be cross-linked. I guess what I'm thinking is, from a technical point of view NPM is a "package thing", but from an app developer point of view, it's "how do I use NPM packages in my app". I think this should be in the app structure guide, going to comment there. It will probably say "here's how to use meteorhacks:npm, the way it works is by creating a package for you, so you might want to read the package docs."